rdkPlugins/GPU/source/GpuPlugin.cpp

```diff
@@ -0,0 +1,308 @@
+ */
+std::string GpuPlugin::getGpuCgroupMountPoint()
+{
+    AI_LOG_FN_ENTRY();
+
+    std::string mountPoint;
+
+    // try and open /proc/mounts for scanning the current mount table
+    FILE* procMounts = setmntent("/proc/mounts", "r");
+    if (procMounts == nullptr)
+    {
+        AI_LOG_SYS_ERROR_EXIT(errno, "failed to open '/proc/mounts'");
+        return std::string();
+    }
+
+    struct mntent mntBuf;
+    struct mntent *mnt;
+    char buf[PATH_MAX + 256];
+
+    // loop over all the mounts
+    while ((mnt = getmntent_r(procMounts, &mntBuf, buf, sizeof(buf))) != nullptr)
+    {
+        // skip entries that don't have a mountpoint, type or options
+        if (!mnt->mnt_dir || !mnt->mnt_type || !mnt->mnt_opts)
+            continue;
+
+        // skip non-cgroup mounts
+        if (strcmp(mnt->mnt_type, "cgroup") != 0)
+            continue;
+
+        // check for the cgroup type
+        char *mntopt = hasmntopt(mnt, "gpu");
+        if (!mntopt || strcmp(mntopt, "gpu") != 0)
+        {
+            continue;
+        }
+
+        AI_LOG_DEBUG("found gpu cgroup, mounted @ '%s'", mnt->mnt_dir);
+
+        mountPoint = mnt->mnt_dir;
+        break;
+    }
+
+    // close /proc/mounts
+    endmntent(procMounts);
+
+    AI_LOG_FN_EXIT();
+    return mountPoint;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Called in the mount namespace of the container
+ *
+ *  crun mounts the gpu cgroup in the container, but it's the root of the
+ *  cgroup tree rather than the cgroup created for the container.
+ *
+ *  For example this is the mount layout setup by crun:
+ *
+ *      /sys/fs/cgroup/cpu/<ID>    -> <ROOTFS>/sys/fs/cgroup/freezer
+ *      /sys/fs/cgroup/memory/<ID> -> <ROOTFS>/sys/fs/cgroup/memory
+ *      /sys/fs/cgroup/gpu         -> <ROOTFS>/sys/fs/cgroup/gpu
+ *      ...
+ *
+ *  We want it to be:
+ *
+ *      /sys/fs/cgroup/gpu/<ID>    -> <ROOTFS>/sys/fs/cgroup/gpu
+ *      ...
+ *
+ *  So we need to replace the existing gpu mount with the cgroup for this
+ *  container.
+ *
+ *  NB: this is not a security thing, but if we don't do this, the app inside
+ *  the container would have to know it's container id to monitor its own usage
+ *  usage. This is a problem for apps which expect the cgroup to be mounted for
+ *  the container only.
+ *
+ *  @param[in]  source      The source path of the cgroup.
+ *  @param[in]  target      The target mount point.
+ *
+ *  @return true if successful, otherwise false.
+ */
+bool GpuPlugin::bindMountGpuCgroup(const std::string &source,
+                                   const std::string &target)
+{
+    AI_LOG_FN_ENTRY();
+
+    // try and do the bind mount
+    if (mount(source.c_str(), target.c_str(), nullptr, MS_BIND, nullptr) < 0)
+    {
+        AI_LOG_SYS_ERROR(errno, "failed to bind mount '%s' to '%s'",
+                         source.c_str(), target.c_str());
+        return false;
+    }
+    else
```

Comment:
Can simplify by removing else statement as we return false inside the above if statement